btf/types.go

```diff
@@ -540,19 +547,19 @@ func (dt *declTag) copy() Type {
 	return &cpy
 }
 
-// typeTag associates metadata with a type.
-type typeTag struct {
+// TypeTag associates metadata with a type.
```

Comment:
Now we're exporting this, could we elaborate in a few sentences what these are used for?